@here/harp-mapview/lib/Utils.ts

```diff
@@ -446,6 +460,229 @@ export namespace MapViewUtils {
+        const image = texture.image;
+        // Assuming RGBA
+        objectSize.heapSize += 4 * image.width * image.height;
+        if (isVisible) {
+            objectSize.gpuSize += 4 * image.width * image.height;
+        }
+    }
+
+    function estimateMaterialSize(
+        material: THREE.Material,
+        isVisible: boolean,
+        objectSize: MemoryUsage
+    ) {
+        if (
+            material instanceof THREE.RawShaderMaterial ||
+            material instanceof THREE.ShaderMaterial
+        ) {
+            const rawMaterial = material;
+            for (const name in rawMaterial.uniforms) {
+                if (rawMaterial.uniforms[name] !== undefined) {
+                    const uniform = rawMaterial.uniforms[name];
+                    if (uniform instanceof THREE.Texture) {
+                        estimateTextureSize(uniform, isVisible, objectSize);
+                    }
+                }
+            }
+        } else if (
+            material instanceof THREE.MeshBasicMaterial ||
+            material instanceof MapMeshBasicMaterial
+        ) {
+            const meshMaterial = material;
+            estimateTextureSize(meshMaterial.map, isVisible, objectSize);
+            estimateTextureSize(meshMaterial.aoMap, isVisible, objectSize);
+            estimateTextureSize(meshMaterial.specularMap, isVisible, objectSize);
+            estimateTextureSize(meshMaterial.alphaMap, isVisible, objectSize);
+            estimateTextureSize(meshMaterial.envMap, isVisible, objectSize);
+        } else if (material instanceof MapMeshStandardMaterial) {
+            const standardMaterial = material;
+            estimateTextureSize(standardMaterial.map, isVisible, objectSize);
+            estimateTextureSize(standardMaterial.aoMap, isVisible, objectSize);
+            estimateTextureSize(standardMaterial.alphaMap, isVisible, objectSize);
+            estimateTextureSize(standardMaterial.envMap, isVisible, objectSize);
+        } else if (
+            material instanceof THREE.LineBasicMaterial ||
+            material instanceof THREE.LineDashedMaterial
+        ) {
+            // Nothing to be done here
+        } else {
+            logger.warn("estimateMeshSize: unidentified material: ", material);
+        }
+    }
+
+    function estimateAttributeSize(
+        attribute: any,
+        attrName: string,
+        isVisible: boolean,
+        objectSize: MemoryUsage
+    ) {
+        let bytesPerElement = 4;
+        let attrBytes = 0;
+        if (attribute.array.BYTES_PER_ELEMENT !== undefined) {
+            bytesPerElement = attribute.array.BYTES_PER_ELEMENT;
+        }
+        if (attribute instanceof THREE.InterleavedBufferAttribute) {
+            const interAttr = attribute;
+            if (interAttr.offset !== 0) {
+                // Array is shared between attributes, count it only once
+                return;
+            }
+            attrBytes = bytesPerElement * attribute.count * attribute.itemSize;
```

Comment:
> `* attribute.itemSize`

Shouldn't it be `attribute.data.stride` ?

Reply:
It looks pretty much like THREE.JS is doing it, no?:

	/**
	 * @param {Array<THREE.BufferGeometry>} geometry
	 * @return {number}
	 */
	estimateBytesUsed: function ( geometry ) {

		// Return the estimated memory used by this geometry in bytes
		// Calculate using itemSize, count, and BYTES_PER_ELEMENT to account
		// for InterleavedBufferAttributes.
		var mem = 0;
		for ( var name in geometry.attributes ) {

			var attr = geometry.getAttribute( name );
			mem += attr.count * attr.itemSize * attr.array.BYTES_PER_ELEMENT;

		}

		var indices = geometry.getIndex();
		mem += indices ? indices.count * indices.itemSize * indices.array.BYTES_PER_ELEMENT : 0;
		return mem;

	},

Reply:
Yes, but this:
```
if (interAttr.offset !== 0) {
```
causes that we count only first attribute, so the first one should report memory of whole buffer.

THREE.Js is iterating over all the attributes.
...

BTW, why are we writing all this stuff by ourselves, if we have something is already in THREE.JS ?
Is it so unusable ?

Reply:
Fixed the issue with counting only the first usage. I made some wrong assumptions... 

The function estimateBytesUsed is only implemented for Geometries, which is only a small fraction of the features we would like to know about.